include/deal.II/base/utilities.h

```diff
@@ -1357,29 +1331,17 @@ namespace Utilities
       }
     else
       {
-        std::string decompressed_buffer;
-
-        // first decompress the buffer
-        (void)allow_compression;
+        // decompress the buffer section into the object
+        boost::iostreams::filtering_istream in;
 #ifdef DEAL_II_WITH_ZLIB
         if (allow_compression)
-          {
-            boost::iostreams::filtering_ostream decompressing_stream;
-            decompressing_stream.push(boost::iostreams::gzip_decompressor());
-            decompressing_stream.push(
-              boost::iostreams::back_inserter(decompressed_buffer));
-            decompressing_stream.write(&*cbegin, std::distance(cbegin, cend));
-          }
-        else
+          in.push(boost::iostreams::gzip_decompressor());
+#else
+        (void)allow_compression;
 #endif
-          {
-            decompressed_buffer.assign(cbegin, cend);
-          }
+        in.push(boost::iostreams::array_source(&*cbegin, &*cend));
 
-        // then restore the object from the buffer
-        std::istringstream              in(decompressed_buffer);
         boost::archive::binary_iarchive archive(in);
-
         archive >> unpacked_object;
```

Comment:
All tests accomplish, but not `base/utilities_pack_unpack_(05|06)`: they fail in this particular line complaining about the input stream:
```
terminate called after throwing an instance of 'boost::archive::archive_exception'
  what():  input stream error
```

My guess is that `boost` expects metainformation about the array in the deserialization process, which we probably do not write out in the serialization process. The metainformation may vanish while writing the contents in the legacy `string`. Calling `boost::serialization::make_array` somewhere might do the trick, and I'm still trying to figure out where.

Please be patient with this PR :)

Reply:
I've found the issue: it's a synchronization problem. The `zlib_compressor` is not flushable, thus we exit the `pack` function before finishing to write into the buffer and we always return that `0` bytes have been written. In that specific test, we rely on the returned number of bytes to recover the serialized object from the buffer, and there we have the issue.

This seems to be an issue that has been around for decades (see [here](https://groups.google.com/g/boost-list/c/0u584AOTIqM)), and was recently proposed as a change to boost https://github.com/boostorg/iostreams/pull/116.

Destroying the stream object before the return statement is supposed to do the trick. I'll give that one a shot. That took a lot of time to figure out...

Reply:
Two hours of debugging for a pair of braces...